pkg/util/stream/publisher_test.go

```diff
@@ -140,16 +140,38 @@ func TestAlwaysClosed(t *testing.T) {
 // Test that unsubscribing basically works.
 func TestUnsubscribe(t *testing.T) {
 	t.Parallel()
+	var wg sync.WaitGroup
+	defer wg.Wait()
 	input := make(chan string)
 	p := NewStringValuePublisher(input, "init")
-	s1 := p.Subscribe(make(chan string, 2))
+
+	s1 := p.Subscribe() // sends "init"
+	unsubscribed := make(chan struct{})
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		if val := <-s1.Chan(); val != "init" {
+			t.Error("dropped first value, got", val)
+		}
+		if val := <-s1.Chan(); val != "hello" {
+			t.Error("dropped second value, got", val)
+		}
+		s1.Unsubscribe()
+		close(unsubscribed)
+
+		// No more messages should be received, but we can't wait forever to verify that.
+		time.Sleep(20 * time.Millisecond)
+		select {
```

Comment:
I didn't originally plan on enforcing that ordering because the original didn't have that ordering. But since you point it out, that does make a lot of sense.